packages/server/src/fhir/search.ts

```diff
@@ -1010,13 +1011,20 @@ function buildChainedSearch(selectQuery: SelectQuery, resourceType: string, para
     throw new OperationOutcomeError(badRequest('Search chains longer than three links are not currently supported'));
   }
 
-  if (getConfig().chainedSearchWithReferenceTables) {
+  if (usesReferenceLookupTable()) {
     return buildChainedSearchUsingReferenceTable(selectQuery, resourceType, param);
   } else {
     return buildChainedSearchUsingReferenceStrings(selectQuery, resourceType, param);
   }
 }
 
+function usesReferenceLookupTable(): boolean {
+  return !!(
+    getConfig().chainedSearchWithReferenceTables ||
+    getAuthenticatedContext().project.features?.includes('reference-lookups')
```

Comment:
I don't think we can use `getAuthenticatedContext()` here.  It works for user `Repository` but would not be valid for `getSystemRepo()`.

Probably needs to be part of `RepositoryContext`?